homeassistant/components/shelly/coordinator.py

```diff
@@ -98,6 +104,7 @@ def __init__(
         self.entry = entry
         self.device = device
         self.device_id: str | None = None
+        self.finish_setup_platforms: list[Platform] | None = None
```

Comment:
It seems like this is always known in advance. I think it would be less error prone if this was a required var for constructing the object even if we need to explicitly pass `None`.

Reply:
Maybe `pending` instead of `finish`?

Reply:
Alternative would be to require it on `async_setup` instead

Reply:
I went with the `async_setup` alternative since we have 4 coordinators but only the ones with `async_setup` (2 out of 4) are using it.